src/documents.c

```diff
@@ -77,6 +77,12 @@ register_document (const char *uri,
 
   file = g_file_new_for_uri (uri);
   path = g_file_get_path (file);
+  if (path == NULL)
+    {
+      g_set_error (error, G_IO_ERROR, G_IO_ERROR_NOT_SUPPORTED,
+                   "URI %s not supported", uri);
```

Comment:
I agree this looks good. We could make the error message more specific, though:

```suggestion
                   "URI %s not supported by the document portal", uri);
```

(It's also quite unfortunate to see xdg-desktop-portal is not translating GError messages, since these are likely to wind up displayed in UIs. But that seems to be a project convention, and consistency is most important, so I won't ask you to change that.)